src/services/ghost/GhostProvider.ts

```diff
@@ -286,64 +284,137 @@ export class GhostProvider {
 			await this.reload()
 		}
 
-		const { response, cost, inputTokens, outputTokens, cacheWriteTokens, cacheReadTokens } =
-			await this.model.generateResponse(systemPrompt, userPrompt)
-
-		this.updateCostTracking(cost)
-
-		TelemetryService.instance.captureEvent(TelemetryEventName.LLM_COMPLETION, {
-			taskId: this.taskId,
-			inputTokens,
-			outputTokens,
-			cacheWriteTokens,
-			cacheReadTokens,
-			cost,
-			service: "INLINE_ASSIST",
-		})
-
-		if (this.isRequestCancelled) {
-			return
-		}
-
-		// First parse the response into edit operations
-		this.suggestions = await this.strategy.parseResponse(response, context)
-		if (this.isRequestCancelled) {
-			this.suggestions.clear()
+		console.log("system", systemPrompt)
+		console.log("userprompt", userPrompt)
+
+		// Initialize the streaming parser
+		this.strategy.initializeStreamingParser(context)
+
+		let hasShownFirstSuggestion = false
+		let totalCost = 0
+		let totalInputTokens = 0
+		let totalOutputTokens = 0
+		let totalCacheWriteTokens = 0
+		let totalCacheReadTokens = 0
```

Comment:
When I see "total" here, I think this is the "lifetime" totals of all completions, but this is just for the current completion right? Maybe we just remove the `total` prefix here?